### PR TITLE
Fix for unit tests for #13

### PR DESCRIPTION
**Pull request by dstockto**
This PR contains a couple of potential fixes for PR #13:

The logic around splitting up the body was previously "working" but I think due to coincidence. Stripping off the leading and trailing pipe in the API response will result in a false if the body is blank or too small, substr will return false. I updated to explicitly test for this. If it was too small then the new $temp becomes a blank array.

The previous logic just exploded the body into $this->data, but with the new field list the logic has changed. Continually calling array_shift on a variable that starts as a blank array will result in nulls. Previously the first call ended up with an empty string due to exploding on false resulting in an array containing a single blank string.

Since we're dealing with (the happy path) strings, I added a bit to ignore nulls and not add the fields if the value is null. This part can likely be improved.

The final two issues were from notices popping up with the blank body test but expecting there would be values in the response for Response Code and AVS Response. I wrapped these in "isset" calls and set them to null if they don't have values. 

I'm completely unfamilar with authorize.net other than what I learned from glancing at a couple of the tests so please make sure I haven't broken any logic. 

TL;DR
This should fix the test/build for PR #13. @acicali

### CODE REVIEW

**Comment by dstockto** [comment]
Just a comment here as I'm getting a little more familiar with what I think is happening. It would seem that this PR (and #13 ) are BC breaks since they are changing the values returned by getCode() and getAVSCode(). Instead of returning a number or letter, they are now returning full words or phrases. If I had code relying on that, it would all be broken. Perhaps a new getter could be introduced that would allow the text representation of these codes to be retrieved from a different method so that existing users of this package would not end up having code breaking if they took this patch?

**Comment by kayladnls** [comment]
@dstockto yes, I agree, that is a BC break and should be avoided. What about `getAVSMessage()` and `getCodeMessage()` or something similar for new functions, that would return that string?

**Comment by dstockto** [comment]
@kayladnls I've pushed up new methods to retrieve the message for code and Avs and put the parsing back in place so that code and avs code are still fetchable.

**Comment by delatbabel** [comment]
Sorry about the delay on this -- I'm slowly finding these PRs and picking them up. Is this still valid? It looks OK to merge but I'll wait to hear back from the original author before doing so.

**Comment by dstockto** [comment]
It's valid from my perspective. Not that that matters much though. :)
